dev-java/mvel/files/mvel-2.3.2-ignore-failing.tests.patch

```diff
@@ -0,0 +1,36 @@
+From 5cb2a222e73b50007117bf96718c67d09e8f46e2 Mon Sep 17 00:00:00 2001
+From: Yuan Liao <liaoyuan@gmail.com>
+Date: Sun, 19 Dec 2021 20:13:59 -0800
+Subject: [PATCH] src/test/java/**: Ignore tests that fail even when run by
+ Maven
+
+Signed-off-by: Yuan Liao <liaoyuan@gmail.com>
+---
+ src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java b/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
+index ea7661bb..1c92c327 100644
+--- a/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
++++ b/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
+@@ -3431,7 +3431,7 @@ public class CoreConfidenceTests extends AbstractTest {
+     assertTrue(result);
+   }
+ 
+-  public void testMapAccessWithNestedMethodCall() {
++  public void noTestMapAccessWithNestedMethodCall() {
```

Comment:
though it is just a matter of style, `@Ignore` annotation is the right way to ignore tests in `junit-4`.

Reply:
That does not seem to work in this particular case:
```diff
diff --git a/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java b/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
index ea7661bb..1c92c327 100644
--- a/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
+++ b/src/test/java/org/mvel2/tests/core/CoreConfidenceTests.java
@@ -3431,7 +3431,8 @@ public class CoreConfidenceTests extends AbstractTest {
     assertTrue(result);
   }
 
+  @org.junit.Ignore
   public void testMapAccessWithNestedMethodCall() {
     String str = "map[aMethod(1)] == \"one\"";
 
     ParserConfiguration pconf = new ParserConfiguration();
@@ -3446,7 +3447,8 @@ public class CoreConfidenceTests extends AbstractTest {
     assertTrue(result);
   }
 
+  @org.junit.Ignore
   public void testMapAccessWithNestedProperty() {
     String str = "map[key] == \"one\"";
 
     ParserConfiguration pconf = new ParserConfiguration();
```

<details><summary>JUnit log</summary>

```
There were 2 failures:
1) testMapAccessWithNestedMethodCall(org.mvel2.tests.core.CoreConfidenceTests)
[Error: unable to resolve method: java.util.HashMap.aMethod() [arglength=1]]
[Near : {... aMethod(1) ....}]
                    ^
[Line: 1, Column: 8]
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.getMethod(ASMAccessorOptimizer.java:1975)
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.compileAccessor(ASMAccessorOptimizer.java:845)
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.optimizeAccessor(ASMAccessorOptimizer.java:281)
	at org.mvel2.ast.ASTNode.optimize(ASTNode.java:159)
	at org.mvel2.ast.ASTNode.getReducedValueAccelerated(ASTNode.java:115)
	at org.mvel2.compiler.ExecutableAccessor.getValue(ExecutableAccessor.java:42)
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.getCollectionPropertyAO(ASMAccessorOptimizer.java:1550)
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.compileAccessor(ASMAccessorOptimizer.java:848)
	at org.mvel2.optimizers.impl.asm.ASMAccessorOptimizer.optimizeAccessor(ASMAccessorOptimizer.java:281)
	at org.mvel2.ast.ASTNode.optimize(ASTNode.java:159)
	at org.mvel2.ast.ASTNode.getReducedValueAccelerated(ASTNode.java:115)
	at org.mvel2.ast.BinaryOperation.getReducedValueAccelerated(BinaryOperation.java:117)
	at org.mvel2.compiler.ExecutableAccessor.getValue(ExecutableAccessor.java:42)
	at org.mvel2.MVEL.executeExpression(MVEL.java:953)
	at org.mvel2.tests.core.CoreConfidenceTests.testMapAccessWithNestedMethodCall(CoreConfidenceTests.java:3446)
2) testMapAccessWithNestedProperty(org.mvel2.tests.core.CoreConfidenceTests)
[Error: unable to resolve method: org.mvel2.tests.core.CoreConfidenceTests$POJO.key() [arglength=0]]
[Near : {... key ....}]
             ^
[Line: 1, Column: 1]
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getMethod(ReflectiveAccessorOptimizer.java:1095)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getMethod(ReflectiveAccessorOptimizer.java:1086)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getMethod(ReflectiveAccessorOptimizer.java:1003)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getBeanProperty(ReflectiveAccessorOptimizer.java:693)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getBeanPropertyAO(ReflectiveAccessorOptimizer.java:488)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.compileGetChain(ReflectiveAccessorOptimizer.java:393)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.optimizeAccessor(ReflectiveAccessorOptimizer.java:163)
	at org.mvel2.ast.ASTNode.optimize(ASTNode.java:159)
	at org.mvel2.ast.ASTNode.getReducedValueAccelerated(ASTNode.java:115)
	at org.mvel2.compiler.ExecutableAccessor.getValue(ExecutableAccessor.java:38)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.getCollectionPropertyAO(ReflectiveAccessorOptimizer.java:860)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.compileGetChain(ReflectiveAccessorOptimizer.java:399)
	at org.mvel2.optimizers.impl.refl.ReflectiveAccessorOptimizer.optimizeAccessor(ReflectiveAccessorOptimizer.java:163)
	at org.mvel2.ast.ASTNode.optimize(ASTNode.java:159)
	at org.mvel2.ast.ASTNode.getReducedValueAccelerated(ASTNode.java:115)
	at org.mvel2.ast.BinaryOperation.getReducedValueAccelerated(BinaryOperation.java:117)
	at org.mvel2.compiler.ExecutableAccessor.getValue(ExecutableAccessor.java:42)
	at org.mvel2.MVEL.executeExpression(MVEL.java:953)
	at org.mvel2.tests.core.CoreConfidenceTests.testMapAccessWithNestedProperty(CoreConfidenceTests.java:3462)

FAILURES!!!
Tests run: 1102,  Failures: 2
```
</details>

Reply:
Resolved as in 8d219d4ab1eea10a33c2ebe73de9653bac3b62bf